vcr/request.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, method, uri, body, headers):
             self.body = body.read()
         else:
             self.body = body
-        self.headers = {}
-        for key in headers:
-            self.add_header(key, headers[key])
+        self.headers = CaseInsensitiveDict()
```

Comment:
The reason to call `.add_header()` is to preserve the behavior of extracting the first element from a list of header values. Re-examining that behavior might be something for a future excursion but it seems orthogonal to this PR. :-)